### ISSUE TITLE
We currently filter short sections as part of the getUnclassifiedSections call

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-server#21]
There's is no need to do this as part of a REST API call and slow it down. It should be done as part of batch processing, presumably while loading data from moves.

**Comment by zacatac** [comment]
***Proposed solution***

**Database schema**
Section now has a new boolean field called 'filter'. This field defines whether or not the section should be filtered in the call to queryUnclassifiedSections 
```python
newSec['filter'] = True/False
```

**~Frontend**
When a client queries for unclassifiedSections a call is made to `queryUnclassifiedSections`, but the call to `filter_UnclassifiedSections` is no longer needed. In the `queryUnclassifiedSections` we filter out any section entry that is labeled with `section['filter'] = True`.

**Backend**
When new sections come from the moves database we will call a modified version of `filter_UnclassifiedSections` that will return `True/False` based on whether or not the section should be filtered on the next call to `'/tripManager/getUnclassifiedSections'`.

**Testing**
Unit tests for both the frontend and backend changes described above.


Contributing: @zacatac @wuhandson @liyeungjohn

**Comment by shankari** [comment]
> ##Backend##
> When new sections come from the moves database we will call a modified version of 
Just to make sure that you have thought through the design sufficiently, what will you do with the `True/False` result?

##~Frontend##
Is this the front end or is the phone code the front end? This is an example of how there is backend and back backend, or in other words, layered architectures are popular.

**Comment by zacatac** [comment]
We will use the `True/False` result to filter the results that we get from `queryUnclassifiedSections` by adding a parameter to the query found in `queryUnclassifiedSections`: 

```python
defaultQueryList = [ {'source':'Shankari'},
                         {'user_id':user_uuid},
                         {'predicted_mode': { '$exists' : True } },
                         {'confirmed_mode': ''},
                         { 'type': 'move' },
                         {'section_end_datetime': {"$gt": weekago}}]
```

As for existing trips we would have to look into adding the `filter` to all sections currently in the database. This may not be the best way, but we couldn't come up with another way to memoize the trips that are labeled as being to short. 

As for the frontend label, I wasn't sure what to label the distinction. 

Thanks for the feedback

**Comment by shankari** [comment]
> As for existing trips we would have to look into adding the filter to all sections currently in the database. > This may not be the best way, but we couldn't come up with another way to memoize the trips as being > too short. 

That seems like a reasonable solution. Please add a small script to do this to your pull request. I typically create such one-off scripts in `CFC_WebApp/utils/migrations`.

> As for the frontend label, I wasn't sure what to label the distinction

My feedback about this was intended as an application of the concepts from lecture, not a critique. Your meaning was perfectly clear and it is indeed hard to come up with a name for it given the current unlayered architecture.

If we do have better layering, like in the last slide of the lecture, then every layer has a name, and this could be easier to talk about.

**Comment by zacatac** [comment]
**Testing**
In 'TestTripManager.py'
'testQueryUnclassifiedSectionsFiltered': tests whether queryUnclassified sections correctly returns sections that are not meant to be filtered, and does not return sections that are meant to be filtered. 

'testQueryUnclassifiedSectionsNoFilterLabel': 
Tests to ensure that if a section is queried that does not have a filter label, then we can immediately assign that piece of data a filter label based on its data. This feature loosens the hard constraint that ALL sections in the database MUST have the filter field set.

In 'TestMovesCollect.py'
'testApplyFilter':
Tests that incoming section data parsed by 'collect.py' is properly labeled with a filter flag based on a previously defined filtering function.

**Comment by zacatac** [comment]
Resolved with #79
